### PR TITLE
Fs-2203-remove-magic-link-auth-roles

### PR DESCRIPTION
**Pull request by gio-karunakaran**
Link to Ticket on Jira: https://digital.dclg.gov.uk/jira/browse/FS-2203

Solution included by this PR:

Have now separated the method to create auth tokens (one for magic links and another for SSOs) the SSO auth method is the same as the original but the Magic Link method checks for an envvar called  ALLOW_ASSESSMENT_LOGIN_VIA_MAGIC_LINK, if this is True it will set the user roles in the token else it will set these roles as an empty list (no roles in token). 

This envvar is set to False in production and unit test configs but True everywhere else. This is done to ensure users have roles unset when authenticating via magic link in production (and unittest so we can accurately test behavior) but allows us to carry on using our test users as normal in development/ test.

### CODE REVIEW

**Comment by gidsg** [comment]
Please add a description to the pull request.

File: config/envs/default.py [hunk 1]
```diff
@@ -99,6 +99,7 @@ class DefaultConfig(object):
     FSD_SESSION_TIMEOUT_SECONDS = CommonConfig.FSD_SESSION_TIMEOUT_SECONDS
     WTF_CSRF_TIME_LIMIT = CommonConfig.WTF_CSRF_TIME_LIMIT
     CREATE_APPLICATION_ON_ACCOUNT_CREATION = False
+    SESSION_ROLES = True
```
Comment (gidsg) [inline config/envs/default.py:102]: Can we call this something more descriptive to the actual use case e.g.
```
ALLOW_ASSESSMENT_LOGIN_VIA_MAGIC_LINK
```
or similar?

Comment (gidsg) [inline config/envs/default.py:102]: Can we add a `getenv` call to these so they are overwritable in the environment?

Comment (gio-karunakaran) [reply config/envs/default.py:102]: done!

File: api/session/auth_session.py [hunk 2]
```diff
@@ -158,3 +158,71 @@ def create_session_details_with_token(
...
+            )
+            current_app.logger.info(
+                f"User logged in to account : {account.id}"
+            )
+            return response
+        except SessionCreateError as e:
+            error_response(404, str(e))
+
+    @classmethod
+    def create_session_details_with_token_via_magic_link(
```
Comment (gidsg) [inline api/session/auth_session.py:203]: Rather than copying the whole function can we pass a parameter to it instead and vary based on the auth method?>

Comment (gio-karunakaran) [reply api/session/auth_session.py]: passing an argument to the function now that determines whether it's being called via magic-link routes.py 👍🏾

File: config/envs/default.py [hunk 3]
```diff
@@ -99,6 +99,9 @@ class DefaultConfig(object):
     FSD_SESSION_TIMEOUT_SECONDS = CommonConfig.FSD_SESSION_TIMEOUT_SECONDS
     WTF_CSRF_TIME_LIMIT = CommonConfig.WTF_CSRF_TIME_LIMIT
     CREATE_APPLICATION_ON_ACCOUNT_CREATION = False
+    ALLOW_ASSESSMENT_LOGIN_VIA_MAGIC_LINK = getenv(
+        "ALLOW_ASSESSMENT_LOGIN_VIA_MAGIC_LINK", True
```
Comment (gidsg) [inline config/envs/default.py:103]: need to use `strtobool` here as string from env will always be considered truthy.

Comment (gio-karunakaran) [reply config/envs/default.py]: thank you! have added this now 👍🏾

**Review by gidsg — APPROVED** [review]
LGTM ✨ 👍

**Pull request merged**